### PR TITLE
archetype keymap for jj50

### PR DESCRIPTION
**Pull request by idimitrakopoulos**
## Description

I added a new keymap for jj50 since there was only the default one

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jj50/keymaps/archetype/config.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
```
Comment (mechmerlin) [inline keyboards/jj50/keymaps/archetype/config.h]: This include guard can be replaced by

`#pragma once`

File: keyboards/jj50/keymaps/archetype/rules.mk [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+F_CPU = 12000000
+
+# Bootloader
+#     This definition is optional, and if your keyboard supports multiple bootloaders of
+#     different sizes, comment this out, and the correct address will be loaded
+#     automatically (+60). See bootloader.mk for all options.
+BOOTLOADER = bootloadHID
+
+# build options
+BOOTMAGIC_ENABLE = yes
```
Comment (mechmerlin) [inline keyboards/jj50/keymaps/archetype/rules.mk]: Unless you're using specific bootmagic functions that allow you to swap keys, I would recommend just setting this value to `lite`

```suggestion
BOOTMAGIC_ENABLE = lite
```

File: keyboards/jj50/keymaps/archetype/keymap.c [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+You should have received a copy of the GNU General Public LicensezZZ
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
+#include "rgblight.h"
+#include "quantum.h"
+
+#define ______ KC_TRNS
```
Comment (mechmerlin) [inline keyboards/jj50/keymaps/archetype/keymap.c]: By including `#include QMK_KEYBOARD_H`, you get access to two `#defines`. They are a 7 space `KC_TRNS` and a `7X KC_NO`. Unless you have specific use cases for the 6 space `KC_TRNS` like you defined here, I'd prefer you use the predefined ones. 

```
_______ = KC_TRNS
XXXXXXX = KC_NO
```

Comment (idimitrakopoulos) [reply keyboards/jj50/keymaps/archetype/keymap.c]: You're right didn't know that was included there. I have removed my defines to make the code simpler

File: keyboards/jj50/keymaps/archetype/rules.mk (hunk 2 above)
Comment (idimitrakopoulos) [reply keyboards/jj50/keymaps/archetype/rules.mk]: sure!

File: keyboards/jj50/keymaps/archetype/config.h (hunk 1 above)
Comment (idimitrakopoulos) [reply keyboards/jj50/keymaps/archetype/config.h]: not sure what I should do here but I will search

Comment (idimitrakopoulos) [reply keyboards/jj50/keymaps/archetype/config.h]: ok i think i fixed it now

**Pull request merged**